src/polymer/iron-flex-layout-classes.ts

```diff
@@ -0,0 +1,246 @@
+        '.layout.no-wrap, .layout.center, .layout.center-center, ' +
+        '.layout.center-justified, .flex, .flex-auto, .flex-none')
+  },
+  {
+    name: 'iron-flex-reverse',
+    selector: elementSelectorToPredicate(
+        '.layout.horizontal-reverse, .layout.vertical-reverse, ' +
+        '.layout.wrap-reverse')
+  },
+  {
+    name: 'iron-flex-alignment',
+    selector: elementSelectorToPredicate(
+        '.layout.start, .layout.center, .layout.center-center, .layout.end, ' +
+        '.layout.baseline, .layout.start-justified, .layout.center-justified, ' +
+        '.layout.center-center, .layout.end-justified, .layout.around-justified, ' +
+        '.layout.justified, .self-start, .self-center, .self-end, .self-stretch, ' +
+        '.self-baseline, .layout.start-aligned, .layout.end-aligned, ' +
+        '.layout.center-aligned, .layout.between-aligned, .layout.around-aligned')
+  },
+  {
+    name: 'iron-flex-factors',
+    // Purposefully skip `.flex` as it's already defined in the iron-flex module.
+    selector: elementSelectorToPredicate(
+        '.flex-1, .flex-2, .flex-3, .flex-4, .flex-5, .flex-6, .flex-7, ' +
+        '.flex-8, .flex-9, .flex-10, .flex-11, .flex-12')
+  },
+  {
+    name: 'iron-positioning',
+    // Purposefully skip `[hidden]` as it's too generic as selector.
+    selector: elementSelectorToPredicate(
+        '.block, .invisible, .relative, .fit, body.fullbleed, ' +
+        '.scroll, .fixed-bottom, .fixed-left, .fixed-top, .fixed-right')
+  }
+];
+
+const isStyleInclude = p.AND(p.hasTagName('style'), p.hasAttr('include'));
+
+class IronFlexLayoutClasses extends HtmlRule {
+  code = 'iron-flex-layout-classes';
+  description = stripIndent(`
+    Warns when iron-flex-layout classes are used without including the style modules.
+  `).trim();
+
+  async checkDocument(parsedDocument: ParsedHtmlDocument, document: Document) {
+    const warnings: FixableWarning[] = [];
+
+    this.convertDeclarations(parsedDocument, document, warnings);
+
+    return warnings;
+  }
+
+  convertDeclarations(
+      parsedDocument: ParsedHtmlDocument, document: Document,
+      warnings: FixableWarning[]) {
+    // Search in the dom-modules.
+    for (const domModule of document.getFeatures({kind: 'dom-module'})) {
+      const template = dom5.query(domModule.astNode, p.hasTagName('template'));
+      if (!template) {
+        continue;
+      }
+      const templateContent = treeAdapters.default.getTemplateContent(template);
+      const missingModules = getMissingStyleModules(templateContent);
+      if (!missingModules) {
+        continue;
+      }
+      // TODO(valdrin): update the warning location to be at the spot where
+      // the class is used.
+      const warning =
+          createWarning(parsedDocument, domModule.astNode, missingModules);
+      const styleNode = dom5.query(templateContent, p.hasTagName('style'));
+      if (!styleNode) {
```

Comment:
this could be simplified by just always adding a new style element, even if there are already other style elements in the template, since Polymer handles multiple styles anyways